examples/amm/lib.rs

```diff
@@ -0,0 +1,373 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
```

Comment:
Can you add some documentation explaining the example? E.g a high level overview of what an AMM is. These examples are used as learning resources and should have good documentation throughout